src/lycophron/tasks/tasks.py

```diff
@@ -6,218 +6,126 @@
-        db.update_record(record)
-        return record
-
-    def _upload_failed(record, text=None):
-        record.status = RecordStatus.FILE_FAIL
-        if text:
-            record.response = text
-        db.update_record(record)
-        return record
-
-    session = create_session(token)
-    bucket_url = record.links.get("bucket")
-    if not bucket_url:
-        print(
-            f"Record {record.doi or record.id} failed to upload files. No bucket url found."
-        )
-        return
-
-    if record.status not in (
-        RecordStatus.FILE_FAIL,
-        RecordStatus.DEPOSIT_SUCCESS,
-    ):
-        print(
-            f"Record {record.doi or record.id} status {record.status}: files can't be uploaded."
-        )
-        return
-
-    for file in record.files:
-        file_name = file["filename"]
-        filepath = file["filepath"]
-        full_path = os.path.join(os.getcwd(), filepath)
-
-        failed = False
-        with open(full_path, "rb") as fp:
-            r = session.put(
-                "%s/%s" % (bucket_url, file_name),
-                data=fp,
-                params={},
-                headers={"Content-Type": "application/octet-stream"},
-            )
-            if r.status_code != 200:
-                failed = True
-
-        if failed:
-            _upload_failed(record)
-        else:
-            _upload_success(record)
-
-        fp.close()
+    draft.publish()
+
+@state_transition(frm=RecordStatus.PUBLISHED, to=RecordStatus.COMMUNITIES_ADDED)
+def add_to_community(client, record, published_record=None):
+    # TODO Return if not to be added to Communities
+    if not published_record:
+        published_record = client.records(record.upload_id).get()
+    # TODO implement add to community, and use Community Table
+
+@app.task
+def process_record(record_id):
+    db_record = db.get_record(record_id)
+    while True:
+        try:
+            # TODO Get from config
+            base_url = "https://127.0.0.1:5000/api"
+            token = "ZxRS6YdOcSabhcwOG3a3PRLzsRZPUcN14w1ReiTisvLAtCKqhTghdmzFthvq"
+
+            client = InvenioAPI(base_url=base_url, access_token=token)
+            client.session.verify = False
+            draft = create_draft_record(client, db_record)
+            update_draft_metadata(client, db_record, draft=draft)
+            upload_record_files(client, db_record, draft=draft)
+            publish_record(client, db_record, draft=draft)
+            # add_to_community(client, db_record)
+            break
+        except HTTPError as e:
+            print(e)
+            if e.response.status_code == 429:
+                sleep(10)
+                continue
+            break
+            # continue
+
+
+def record_dispatcher():
+    records = db.get_unpublished_deposits(10)
+    for record in records:
+        record.status = RecordStatus.QUEUED
+        db.session.commit()
+        process_record.apply_async(args=[record.id])
```

Comment:
Dispatcher code needs to move to `app.py` probably, and we need to find the best way to start the workers along with it.